lib/miq_expression.rb

```diff
@@ -625,10 +625,14 @@ def sql_supports_atom?(exp)
       # => TODO: support count of child relationship
       return false if exp[operator].key?("count")
 
-      return field_in_sql?(exp[operator]["field"])
+      return field_in_sql?(exp[operator]["field"]) && value_in_sql?(exp[operator]["value"])
     end
   end
 
+  def value_in_sql?(value)
+    !Field.valid_field?(value) || Field.parse(value).attribute_supported_by_sql?
```

Comment:
if you put a real attribute or a virtual attribute on the right hand side, then that is supported by sql.
if you put a static string (or number) on the right hand side, then that is supported by sql.

If you put a virtual attribute that has no arel, or a dynamic value on the right, then that will not be a supported value in sql